src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -43,33 +43,44 @@ OpBase *NewNodeByLabelScanOp(const ExecutionPlan *plan, const QGNode *n) {
 }
 
 void NodeByLabelScanOp_SetIDRange(NodeByLabelScan *op, UnsignedRange *id_range) {
-	memcpy(op->id_range, id_range, sizeof(UnsignedRange));
+	UnsignedRange_Free(op->id_range);
+	op->id_range = UnsignedRange_Clone(id_range);
```

Comment:
`NodeByLabelScanOp_SetIDRange` gets called at most once, so this seems like unnecessary complexity to me.

At minimum, wouldn't it be preferable to NULL-initialize `op->id_range` in `NewNodeByLabelScanOp` and use some conditional logic in `ConstructIterator` to remove the free here?

Reply:
I didn't like the memcpy approach here suppose one day someone changes `UnsignedRange` struct to contain a pointer to some heap allocation, performing memcpy is dangerous, a `clone` function is a bit more clear and safe, plus the overhead here is negligible.